.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Commitlint'
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+  actions: read
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+      - uses: nrwl/nx-set-shas@v3
+      - name: Monorepo install
+        uses: ./.github/actions/yarn-nm-install
+      - name: Validate current commit (last commit) with commitlint
+        if: github.event_name == 'push'
+        run: npx commitlint --from HEAD~1 --to HEAD --verbose
+      - name: Validate PR commits with commitlint
+        if: github.event_name == 'pull_request'
+        run: npx commitlint --from ${{ github.event.pull_request.head.sha }}~${{ github.event.pull_request.commits }} --to ${{ github.event.pull_request.head.sha }} --verbose
```

Comment:
What benefit do you see running both checks? I think the latter would always cover the former too.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

> Runs your workflow when activity on a pull request in the workflow's repository occurs.

Since the check is quite fast I would only run the latter.

Reply:
Do we ever push to `main`? That might be my only thought? 🤔

Reply:
Good point. If we do the CI would run too late though and it would advise the commiter to alter the history of main which is probably even worse...

Reply:
+1 checking on push would be too late and it might actually report errors when using merge commits when we can't squash.

Reply:
Okay so it sounds like we're all agreed to just incl. PRs. I'll make those changes ☺️